### ISSUE TITLE
PR 2679 causes validation errors in Vulkan CTS

### ISSUE THREAD

**Issue by alegal-arm** [KhronosGroup/SPIRV-Tools#2696]
As of https://github.com/KhronosGroup/SPIRV-Tools/commit/7c294608ca19fbbf7244b36965793a961eeacde5 spir-val started to produce rather cryptic errors for some SPIR-V generated by glslang. An example is below. I have hard times trying to understand what the diagnostics actually wanted to say.

Shader:
```
#version 440 core
precision highp float;
precision highp int;
layout(location = 0) in vec4 var0;
layout(location = 1) in vec4 var1;
layout(location = 0, component = 0) out float dEQP_FragColor_0_0;
layout(location = 0, component = 1) out vec3  dEQP_FragColor_0_1[2];
layout(location = 1, component = 0) out float dEQP_FragColor_1_0;
void main()
{
    vec4 out0 = var0;
    out0 /= vec4(255.0);
    dEQP_FragColor_0_0 = out0.x;
    dEQP_FragColor_0_1[0] = out0.yzw;

    vec4 out1 = var1;
    out1 /= vec4(255.0);
    dEQP_FragColor_1_0 = out1.x;
    dEQP_FragColor_0_1[1] = out1.yzw;
}
```

SPIR-V from glslang and spir-val message
```
Validation FAILED: Component decoration specified for type 26[%_arr_v3float_uint_2] that is not a scalar or vector
  %28 = OpVariable %_ptr_Output__arr_v3float_uint_2 Output

; SPIR-V
; Version: 1.0
; Generator: Khronos Glslang Reference Front End; 7
; Bound: 47
; Schema: 0
               OpCapability Shader
          %1 = OpExtInstImport "GLSL.std.450"
               OpMemoryModel Logical GLSL450
               OpEntryPoint Fragment %4 "main" %11 %18 %28 %36 %40
               OpExecutionMode %4 OriginUpperLeft
               OpDecorate %11 Location 0
               OpDecorate %18 Component 0
               OpDecorate %18 Location 0
               OpDecorate %28 Component 1
               OpDecorate %28 Location 0
               OpDecorate %36 Location 1
               OpDecorate %40 Component 0
               OpDecorate %40 Location 1
       %void = OpTypeVoid
          %3 = OpTypeFunction %void
      %float = OpTypeFloat 32
    %v4float = OpTypeVector %float 4
%_ptr_Function_v4float = OpTypePointer Function %v4float
%_ptr_Input_v4float = OpTypePointer Input %v4float
         %11 = OpVariable %_ptr_Input_v4float Input
  %float_255 = OpConstant %float 255
         %14 = OpConstantComposite %v4float %float_255 %float_255 %float_255 %float_255
%_ptr_Output_float = OpTypePointer Output %float
         %18 = OpVariable %_ptr_Output_float Output
       %uint = OpTypeInt 32 0
     %uint_0 = OpConstant %uint 0
%_ptr_Function_float = OpTypePointer Function %float
    %v3float = OpTypeVector %float 3
     %uint_2 = OpConstant %uint 2
%_arr_v3float_uint_2 = OpTypeArray %v3float %uint_2
%_ptr_Output__arr_v3float_uint_2 = OpTypePointer Output %_arr_v3float_uint_2
         %28 = OpVariable %_ptr_Output__arr_v3float_uint_2 Output
        %int = OpTypeInt 32 1
      %int_0 = OpConstant %int 0
%_ptr_Output_v3float = OpTypePointer Output %v3float
         %36 = OpVariable %_ptr_Input_v4float Input
         %40 = OpVariable %_ptr_Output_float Output
      %int_1 = OpConstant %int 1
          %4 = OpFunction %void None %3
          %5 = OpLabel
          %9 = OpVariable %_ptr_Function_v4float Function
         %35 = OpVariable %_ptr_Function_v4float Function
         %12 = OpLoad %v4float %11
               OpStore %9 %12
         %15 = OpLoad %v4float %9
         %16 = OpFDiv %v4float %15 %14
               OpStore %9 %16
         %22 = OpAccessChain %_ptr_Function_float %9 %uint_0
         %23 = OpLoad %float %22
               OpStore %18 %23
         %31 = OpLoad %v4float %9
         %32 = OpVectorShuffle %v3float %31 %31 1 2 3
         %34 = OpAccessChain %_ptr_Output_v3float %28 %int_0
               OpStore %34 %32
         %37 = OpLoad %v4float %36
               OpStore %35 %37
         %38 = OpLoad %v4float %35
         %39 = OpFDiv %v4float %38 %14
               OpStore %35 %39
         %41 = OpAccessChain %_ptr_Function_float %35 %uint_0
         %42 = OpLoad %float %41
               OpStore %40 %42
         %44 = OpLoad %v4float %35
         %45 = OpVectorShuffle %v3float %44 %44 1 2 3
         %46 = OpAccessChain %_ptr_Output_v3float %28 %int_1
               OpStore %46 %45
               OpReturn
               OpFunctionEnd
```

**Comment by alegal-arm** [comment]
cc: @jozefkucia @alan-baker

**Comment by jozefkucia** [comment]
It should be fixed by #2697.
